lib/core/auth/auth_provider.js

```diff
@@ -1,40 +1,46 @@
 'use strict';
 
-/**
- * Creates a new AuthProvider, which dictates how to authenticate for a given
- * mechanism.
- * @class
- */
+class AuthContext {
```

Comment:
Is this how we will be creating ts-like interfaces moving forward? I haven't seen this style in the codebase so far. I assumed we were doing option types in JSDOC. [Something like this example here](https://jsdoc.app/tags-typedef.html):

```js
/**
 * The complete Triforce, or one or more components of the Triforce.
 * @typedef {Object} WishGranter~Triforce
 * @property {boolean} hasCourage - Indicates whether the Courage component is present.
 * @property {boolean} hasPower - Indicates whether the Power component is present.
 * @property {boolean} hasWisdom - Indicates whether the Wisdom component is present.
 */
```

Reply:
yes, I'll add that now, but in the future expect to get this type information directly from TypeScript. This is likely a type we will introduce early on considering its very small in scope

Reply:
I'd added some type information for you here. In the future type information will be handled by actually using TypeScript, this will likely be an early migrated type since its so small in scope.